### PR TITLE
Make deletion protection consistent across all modules

### PR DESCRIPTION
**Pull request by juliocc**
This PR exposes the `deletion_protection` attributed in GKE clusters introduced in version 5 of the provider.

---
**Checklist**

I applicable, I acknowledge that I have:
- [x] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [x] Ran `terraform fmt` on all modified files
- [x] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [x] Made sure all relevant tests pass

### CODE REVIEW

File: modules/gke-cluster-autopilot/variables.tf [hunk 1]
```diff
@@ -34,6 +34,12 @@ variable "backup_configs" {
   nullable = false
 }
 
+variable "deletion_protection" {
+  description = "Whether or not to allow Terraform to destroy the cluster. Unless this field is set to false in Terraform state, a terraform destroy or terraform apply that would delete the cluster will fail."
+  default     = true
```
Comment (ludoo) [inline modules/gke-cluster-autopilot/variables.tf:39]: we should have it as false by default Julio

Comment (juliocc) [reply modules/gke-cluster-autopilot/variables.tf:39]: the provider defaults to true

Comment (ludoo) [reply modules/gke-cluster-autopilot/variables.tf:39]: well the provider is not always sane :) anyway, we discussed it and I understasnd your point, I would just do the same change on GCS

**Review by ludoo — APPROVED** [review]

File: modules/gke-cluster-autopilot/variables.tf (hunk 1 above)
Comment (ludoo) [reply modules/gke-cluster-autopilot/variables.tf:39]: GCS already has it, I stand corrected; can we bubble up this variable in GKE blueprints too? anf FAST

Comment (juliocc) [reply modules/gke-cluster-autopilot/variables.tf:39]: As discussed, I'll update `cloudsql-instance` which is behaving in the opposite way and expose it in blueprints to allow easy destruction.

**Review by ludoo — APPROVED** [review]

**Comment by juliocc** [comment]
@lcaggio this mostly affects the data blueprints. Can you take a look and make sure I'm not breaking anything?

**Pull request merged**